src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/LambdaBodyLengthCheck.java

```diff
@@ -0,0 +1,234 @@
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * class Test {
+ *   Runnable r = () -&gt; { // violation, 6 lines
+ *       System.out.println(2); // line 2 of lambda
+ *       System.out.println(3);
+ *       System.out.println(4);
+ *       System.out.println(5);
+ *   };
+ *
+ *   Runnable r2 = () -&gt; // violation, 6 lines
+ *       "someString".trim()
+ *                   .trim()
+ *                   .trim()
+ *                   .trim()
+ *                   .trim()
+ *                   .trim();
+ *
+ *   Runnable r3 = () -&gt; { // ok, 5 lines
+ *       System.out.println(2);
+ *       System.out.println(3);
+ *       System.out.println(4);
+ *   };
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code maxLen.lambdaBody}
+ * </li>
+ * </ul>
+ *
+ * @since 8.37
+ */
+@StatelessCheck
+public class LambdaBodyLengthCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "maxLen.lambdaBody";
+
+    /** Default maximum number of lines. */
+    private static final int DEFAULT_MAX = 10;
+
+    /** Specify the maximum number of lines allowed. */
+    private int max = DEFAULT_MAX;
+
+    /**
+     * Setter to specify the maximum number of lines allowed.
+     *
+     * @param length the maximum length of lambda body.
+     */
+    public void setMax(int length) {
+        max = length;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.LAMBDA};
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (ast.getParent().getType() != TokenTypes.SWITCH_RULE) {
+            final int length = getLength(ast);
+            if (length > max) {
+                log(ast, MSG_KEY, length, max);
+            }
+        }
+    }
+
+    private static int getLength(DetailAST ast) {
+        final DetailAST lambdaBody = ast.getLastChild();
+        final int length;
+        if (lambdaBody.getType() == TokenTypes.SLIST) {
+            length = lambdaBody.getLastChild().getLineNo() - lambdaBody.getLineNo();
+        }
+        else {
+            length = getLastNodeLineNumber(lambdaBody) - getFirstNodeLineNumber(lambdaBody);
+        }
+        return length + 1;
+    }
+
+    /**
+     * Get last node line number.
+     *
+     * @param lambdaBody lambda body node.
+     * @return last node line number.
+     */
+    private static int getLastNodeLineNumber(DetailAST lambdaBody) {
+        DetailAST node = lambdaBody.getLastChild();
+        int result = node.getLineNo();
+        while (node != null) {
```

Comment:
Please change this loop to `do { ... } while()`. It is strange to see `node != null` after `node.getLineNo()`.

Reply:
done